analyses/cell-type-ewings/template_notebooks/clustering-workflow/01-clustering-metrics.Rmd

```diff
@@ -0,0 +1,257 @@
+---
+title: "Clustering exploration for `r params$library_id`"
+author: Ally Hawkins
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    toc: true
+    toc_depth: 3
+    code_folding: "hide"
+params:
+  library_id: "SCPCL000822"
+  sce_file: "../../../../data/current/SCPCP000015/SCPCS000490/SCPCL000822_processed.rds"
+  cluster_results_file: "SCPCL000822_clusters.tsv"
+---
+
+This notebook summarizes clustering metrics for `r params$library_id`.
+
+Louvain clusters are calculated using the following range of parameters: 
+
+- Nearest neighbors: 5, 10, 15, 20, 25, 30, 35, and 40
+- Resolution: 0.5, 1, 1.5
+
+Leiden clusters are calcualted using the following range of parameters: 
+
+- Objective function: CPM and modularity 
+- Nearest neighbors: 5, 10, 15, 20, 25, 30, 35, and 40
+- Resolution: 0.5, 1, 1.5
+
+- Metrics are then calculated to evaluate clustering results across all parameters: 
+  - Silhouette width 
+  - Cluster purity 
+  - Cluster stability 
+
+## Setup
+
+```{r}
+# check that sce file exists
+stopifnot(
+  "sce file does not exist" = file.exists(params$sce_file)
+)
+```
+
+
+```{r packages}
+suppressPackageStartupMessages({
+  # load required packages
+  library(SingleCellExperiment)
+  library(ggplot2)
+})
+
+# Set default ggplot theme
+theme_set(
+  theme_classic()
+)
+
+# set seed
+set.seed(2024)
+```
+
+
+```{r base paths}
+# The path to this module
+module_base <- rprojroot::find_root(rprojroot::is_renv_project)
+
+# source in clustering functions
+clustering_functions <- file.path(module_base, "scripts", "utils", "clustering-functions.R")
+source(clustering_functions)
+```
+
+
+```{r}
+# read in input sce files
+sce <- readr::read_rds(params$sce_file)
+```
+
+
+## Calculate clusters
+
+
```

Comment:
I might define a chunk here for parameters, e.g. `nn <- seq(5, 40, 5)`. _Or_ we could actually make these actual Rmd params with the values you use here as defaults, which would allow some flexibility if it's needed when running this as a template across samples.